apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.cs

```diff
@@ -180,6 +195,43 @@ private async Task StopCompletionAsync(Task[] subscriberTasks, HashSet<Task> reg
         }
     }
 
+    /// <summary>
+    /// Gets the delay between the extend requests for automatic message lease extension.
+    /// </summary>
+    /// <param name="ackDeadlineSeconds">The Ack deadline in seconds for which delay will be calculated.</param>
+    /// <returns>The delay between auto-extends.</returns>
+    private TimeSpan GetAutoExtendDelay(int ackDeadlineSeconds)
+    {
+        // The delay is calculated as AckDeadline - AckWindow.
+        var delay = TimeSpan.FromSeconds(ackDeadlineSeconds) - _ackExtensionWindow;
+        return TimeSpan.FromTicks(Math.Max(delay.Ticks, MinimumLeaseExtensionDelay.Ticks));
+    }
+
+    /// <summary>
+    /// Returns the effective Acknowledgement Deadline based on the Maximum and/or Min Acknowledgement Deadline settings provided in the settings.
+    /// </summary>
+    /// <param name="specifiedAckDeadlineSeconds">The Ack deadline in seconds for which effective deadline is to be calculated.</param>
+    /// <param name="exactlyOnceDelivery">A boolean flag indicating if subscription is exactly-once delivery or not.</param>
+    /// <returns>The effective AckDeadline to use.</returns>
+    private TimeSpan GetEffectiveAckDeadline(int specifiedAckDeadlineSeconds, bool exactlyOnceDelivery)
```

Comment:
My feelings earlier on are still present - why would a user say "I want to extend by exactly 10 seconds, but not less than 5 seconds and not more than 20 seconds"? That basically allows them to set *any* extension value, doesn't it, just by setting the min to 0 and the max to infinity.